### PR TITLE
Define Markdownlint GitHub Action and set

### LINKED ISSUE #22: Add GitHub workflow for `markdownlint`

### PR DESCRIPTION
**Pull request by HarmonicReflux**
This Pull request adds a Markdownlinter to GitHub Actions to make sure `*.md ` files are formatted to standard. 

In particular, the following features were cofigured:
- Added `name` fields to the workflow and job for better identification in GitHub Actions UI
- Set workflow triggers to `main` and `develop` branches for both push and pull request events
- Enhanced step descriptions for clarity and easier debugging
- Ensured markdownlint CLI2 automatically fixes issues and targets all Markdown files

Closes #22.

### CODE REVIEW

File: .github/workflows/markdownlint-cli2-action.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    name: Markdown Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      - name: Run markdownlint
+        uses: DavidAnson/markdownlint-cli2-action@v18
+        with:
+          fix: true       
+          globs: '**/*.md'
```
Comment (djmaxus) [inline .github/workflows/markdownlint-cli2-action.yml:24]: Is it possible to not check the spelling in the auto-generated `CHANGELOG.md`?

**Review by djmaxus — CHANGES_REQUESTED** [review]

**Review by djmaxus — APPROVED** [review]
Looks and works great! I'm happy to merge this now. Thank you @HarmonicReflux

**Comment by alexdewar** [comment]
For future reference, I'd be wary about merging someone's PR if they've marked it as draft! That can often mean something isn't finished or it hasn't been tested properly. Not sure if this is the case or not here.

**Comment by djmaxus** [comment]
@alexdewar yes, absolutely! This is a fair point, and I am so sorry for my hurry.
I was confused about whether I accidentally marked it as a draft and didn't double-check.

My apologies, @HarmonicReflux 🙏🏼

**Pull request merged**